lib/common/utils.ts

```diff
@@ -363,3 +363,9 @@ export function patchMethod(target: any, name: string,
   return delegate;
 }
 
+export function makeNamedFnChainable(fn: (self: any, args: any[]) => any) {
```

Comment:
- pls move this fn where it is used (ie `events.ts`).
- use arrow fn,
- wouldn't `callAndReturnFirstParam(...)` be more explicit ?